File: learning/rihal/Demo 2/Demo 2 Class files/FileHandler.py
from abc import ABC, abstractmethod

#abstract class for file upload,split,download,merge,search
class FileHandler(ABC):
    
    @abstractmethod
    def upload_file(self, file_path: str, file_name: str, mimetype: str):
        pass

    @abstractmethod
    def upload_chunk(self, service, chunk_filename: str, mimetype: str, file_name: str, chunk_index: int):
        pass

    @abstractmethod
    def update_metadata(self, metadata: str):
        pass
    
    @abstractmethod
    def download_file(self, file_id:str, save_path:str):
        pass
    
    @abstractmethod
    def search_file():
        pass
    
    @abstractmethod
    def merge_chunks(self,file_paths:str, merged_file_path:str):
        pass
    
    @abstractmethod
    def download_and_merge_chunks(self, file_name:str, save_path:str):
        pass

    @abstractmethod
    def download_from_all_buckets(self, file_name:str, save_path:str):
        pass
    

File: learning/rihal/Mobile App Initial/syncly (final demo)/app/src/main/java/com/example/syncly/FileHandler.java
package com.example.syncly;

public abstract class FileHandler {

    // Abstract method for uploading the full file
    public abstract void uploadFile(String filePath, String fileName, String mimeType);

    // Abstract method to update metadata
    public abstract void updateMetadata(String metadata);

    // Abstract method to search for a file
    public abstract void searchFile();

    //Abstract method to download a file
    public abstract void downloadFile(String fileName, String savePath);
}
